### PR TITLE
Convert to new plugin API

### PR DESCRIPTION
**Pull request by goderbauer**
/cc @collinjackson @mravn-google

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -30,6 +30,7 @@ android {
         disable 'InvalidPackage'
     }
     dependencies {
+        compile 'com.google.firebase:firebase-core:10.2.1'
```
Comment (goderbauer) [inline android/build.gradle]: @collinjackson I can't get it to compile without this...

Comment (collinjackson) [inline android/build.gradle]: OK, I wonder why it worked for me. Looks fine though

File: ios/Classes/FirebaseStoragePlugin.m [hunk 2]
```diff
@@ -18,42 +18,48 @@ - (FlutterError *)flutterError {
 @implementation FirebaseStoragePlugin {
 }
 
-- (instancetype)initWithController:(FlutterViewController *)controller {
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  FlutterMethodChannel *channel =
+  [FlutterMethodChannel methodChannelWithName:@"firebase_storage"
+                              binaryMessenger:[registrar messenger]];
+  FirebaseStoragePlugin *instance = [[FirebaseStoragePlugin alloc] init];
+  [registrar addApplicationDelegate:instance];
```
Comment (collinjackson) [inline ios/Classes/FirebaseStoragePlugin.m]: I don't think we need to do this if we aren't using any UIApplicationDelegate calls.

Comment (goderbauer) [inline ios/Classes/FirebaseStoragePlugin.m]: Good catch! Removed.

File: android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java [hunk 3]
```diff
@@ -29,18 +28,17 @@
  * FirebaseStoragePlugin
  */
 public class FirebaseStoragePlugin implements MethodCallHandler {
-  private FlutterActivity activity;
   private FirebaseStorage firebaseStorage;
 
-  public static FirebaseStoragePlugin register(FlutterActivity activity) {
-    return new FirebaseStoragePlugin(activity);
+  public static void registerWith(Registrar registrar) {
+    final MethodChannel channel = new MethodChannel(registrar.messenger(), "firebase_storage");
```
Comment (mravn-google) [inline android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java]: The channel name needs to be globally unique. In other plugins, we have used `"plugins.flutter.io/plugin_name"`

**Pull request merged**